awscli/customizations/s3/subcommands.py

```diff
@@ -456,6 +456,46 @@
     )
 }
 
+
+TAGGING = {
+    'name': 'tagging', 'cli_type_name': 'map',
+    'schema': {
+        'type': 'map',
+        'key': {'type': 'string'},
+        'value': {'type': 'string'}
+    },
+    'help_text': (
+        'The tag-set for the result object. The tag-set must be encoded '
+        'as URL Query parameters. (For example, "Key1=Value1") '
+        '``--tagging-directive`` option defines merge strategy of the '
+        '``--tagging`` value and the original object tag-set.'
+    )
+}
+
+
+TAGGING_DIRECTIVE = {
+    'name': 'tagging-directive',
```

Comment:
What was the rationale on moving away from `--merge-tagging` or `--merge-tags` and having that be a boolean flag? I'm a little hesitant in using `--tagging-directive` because it shadows the API parameter and has different value and semantics, which could be confusing to customers. Also  the name `--merge-tagging`/`--merge-tags` does a better job of self-documenting what it does then `--tagging-directive`.

Reply:
The main reason was to leave space for other merging strategies if we decide to add them

Reply:
Ah so I was thinking we would model it similar to `--sse` where at first it had an action of `store_true` and then expanded it to allow `nargs` of `?`: https://github.com/aws/aws-cli/blob/9b44abec0b656befa01acea07b92f35ebbbe343a/awscli/customizations/s3/subcommands.py#L177-L178 So for `--merge-tagging` it's presence would equate the logic for this parameter's `merge`. Then when we add other merge strategies, we would switch it to be `nargs=?` and have choices like this (we could figure out good names for them later when we implement them):

1. `override` - This is the default value for `--merge-tagging`
     ```
     aws s3 cp s3://bucket/object s3://bucket/object2 --tagging Key=value --merge-tagging override
     ```
2. `only-missing` - This would do the option of only adding tags that were missing from the source object
     ```
     aws s3 cp s3://bucket/object s3://bucket/object2 --tagging Key=value --merge-tagging only-missing
     ```
3. `only-existing` - This would do the option of only updating tag values if the key exist
     ```
     aws s3 cp s3://bucket/object s3://bucket/object2 --tagging Key=value --merge-tagging only-existing
     ```

How does that sound?

Reply:
Sounds good